azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/conditional/OperatingSystem.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.boot.conditional;
+
+/**
+ * List of available for operating system for condition.
+ *
+ * @author Arthur Gavlyukovskiy
+ */
+public enum OperatingSystem {
```

Comment:
@gavlyukovskiy what about MacOS ? I have seen certain things very different here. For example some UNIX perf counters are collected by parsing /proc/stats file. But MacOS X doesn't have any such file? Also as of now I haven't been able to successfully set up collectd in Mac OS. What should be the way to differentiate this here. Do we need another Constant here for MacOS. This is not very important for now as most server code is never ran on MacOS but rather on Linux/Unix system but people use mac for dev time and can throw misleading information. What is your thought here?

Reply:
Actually I need only `WINDOWS` for specific performance module to place it in conditional, I added `UNIX` for demonstrating purpose. If you will have modules that should be enabled only for Unix, but not for MacOS and vise versa we can add this one later.

Reply:
Yes there is a module Application Insights CollectD exporter module. Look at this class `com.microsoft.applicationinsights.collectd.ApplicationInsightsWriter`

Reply:
As I can see this one is not enabled by default. This conditional is used for cases when you want to enable this module by default, but only for specific OS.